src/nunit.analyzers/Extensions/AttributeArgumentSyntaxExtensions.cs

```diff
@@ -138,45 +145,39 @@ private static bool TryChangeType(ITypeSymbol targetType, object argumentValue)
             }
         }
 
-        private static Type GetTargetReflectionType(ITypeSymbol targetType)
+        private static bool HasBuiltInImplicitConversion(ITypeSymbol argumentType, ITypeSymbol targetType, SemanticModel model)
         {
-            var containingAssembly = targetType.ContainingAssembly ?? targetType.BaseType.ContainingAssembly;
-            string assembly = ", " + containingAssembly.Identity.ToString();
-            string typeName = AttributeArgumentSyntaxExtensions.GetQualifiedTypeName(targetType);
-
-            // First try to get type using assembly-qualified name, and if that fails try to get type
-            // using only the type name qualified by its namespace.
-            // This is a hacky attempt to make it work for types that are forwarded in .NET Core, e.g.
-            // Double which exists in the System.Runtime assembly at design time and in
-            // System.Private.CorLib at runtime, so targetType.ContainingAssembly will denote the wrong
-            // assembly, System.Runtime. See e.g. the following comment
-            // https://github.com/dotnet/roslyn/issues/16211#issuecomment-373084209
-            var targetReflectionType = Type.GetType(typeName + assembly, false) ?? Type.GetType(typeName, false);
-            return targetReflectionType;
+            var conversion = model.Compilation.ClassifyConversion(argumentType, targetType);
+            return conversion.IsImplicit && !conversion.IsUserDefined;
         }
 
-        private static string GetQualifiedTypeName(ITypeSymbol targetType)
+        private static bool CanBeTranslatedByTypeConverter(
+            ITypeSymbol targetTypeSymbol,
+            object argumentValue)
         {
-            // Note that this does not take into account generics,
-            // so if that's ever added to attributes this will have to change.
-            var namespaces = new Stack<string>();
+            var typeName = targetTypeSymbol.GetFullMetadataName();
+            var targetType = IntrinsicTypeConverters.FirstOrDefault(t => t.type.FullName == typeName);
 
-            var @namespace = targetType.ContainingNamespace ?? targetType.BaseType.ContainingNamespace;
-            var typeName = !string.IsNullOrEmpty(targetType.Name) ? targetType.Name : targetType.BaseType.Name;
-
-            while (!@namespace.IsGlobalNamespace)
+            if (targetType.typeConverter == null)
             {
-                namespaces.Push(@namespace.Name);
-                @namespace = @namespace.ContainingNamespace;
+                return false;
             }
 
-            return $"{string.Join(".", namespaces)}.{typeName}";
-        }
+            var typeConverter = targetType.typeConverter.Value;
+            if (typeConverter.CanConvertFrom(argumentValue.GetType()))
+            {
+                try
+                {
+                    typeConverter.ConvertFrom(null, CultureInfo.InvariantCulture, argumentValue);
```

Comment:
Keep in mind that both `CanConvertFrom` and `ConvertFrom` may behave differently depending on which version of .NET Framework or .NET Core the analyzer is executing on. The framework and version that the analyzer is executing on will usually not match the version and framework that the tests execute on.

Reply:
Thanks for the warning. I'll keep that in mind. My plan is to go with this initially, but if we run into differences then we can choose the safer approach and just assume that strings can always be converted into the mentioned types.